From 16d318bf0004732b427db989c84703d2924d4681 Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Tue, 29 Mar 2022 15:22:36 +0530
Subject: [PATCH] Use Device instead of RecoveryUI

Use Device instead of RecoveryUI to resolve build errors.
This change is in correspondence to aosp commit
"Perform data wipe in recovery if ota package has powerwash"

Change-Id: Ic8315673d4c089f2e3ee966ed842bc8428934c3e
Signed-off-by: svenate <salini.venate@intel.com>
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 install/fuse_install_udisk.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/install/fuse_install_udisk.cpp b/install/fuse_install_udisk.cpp
index 4258b99d..5f470543 100644
--- a/install/fuse_install_udisk.cpp
+++ b/install/fuse_install_udisk.cpp
@@ -152,7 +152,7 @@ static bool StartInstallPackageFuse(std::string_view path) {
   return run_fuse_sideload(std::move(fuse_data_provider)) == 0;
 }
 
-static InstallResult InstallWithFuseFromPathUdisk(std::string_view path, RecoveryUI* ui) {
+static InstallResult InstallWithFuseFromPathUdisk(std::string_view path, Device* device) {
   // We used to use fuse in a thread as opposed to a process. Since accessing
   // through fuse involves going from kernel to userspace to kernel, it leads
   // to deadlock when a page fault occurs. (Bug: 26313124)
@@ -188,9 +188,9 @@ static InstallResult InstallWithFuseFromPathUdisk(std::string_view path, Recover
     }
     auto package =
         Package::CreateFilePackage(FUSE_SIDELOAD_HOST_PATHNAME,
-                                   std::bind(&RecoveryUI::SetProgress, ui, std::placeholders::_1));
+                                   std::bind(&RecoveryUI::SetProgress, device->GetUI(), std::placeholders::_1));
     result =
-        InstallPackage(package.get(), FUSE_SIDELOAD_HOST_PATHNAME, false, 0 /* retry_count */, ui);
+        InstallPackage(package.get(), FUSE_SIDELOAD_HOST_PATHNAME, false, 0 /* retry_count */, device);
     break;
   }
 
@@ -239,7 +239,7 @@ InstallResult ApplyFromUdisk(Device* device) {
   ui->Print("\n-- Install %s ...\n", path.c_str());
   SetUdiskUpdateBootloaderMessage();
 
-  auto result = InstallWithFuseFromPathUdisk(path, ui);
+  auto result = InstallWithFuseFromPathUdisk(path, device);
   ensure_path_unmounted(udisk_root);
   return result;
 }
-- 
2.17.1

